cscs-checks/apps/lammps/lammps_check.py

```diff
@@ -7,102 +7,169 @@
+
+}
+
+daint_cpu_large = {
+    'prod': (5310, -0.65, None, 'timesteps/s'),
+}
+
+eiger_cpu_large = {
+    'prod': (6500, -0.10, None, 'timesteps/s'),
+}
+
+pilatus_cpu_large = {
+    'prod': (7500, -0.10, None, 'timesteps/s'),
+}
+
+REFERENCE_CPU_PERFORMANCE_LARGE = {
+    'daint:mc': daint_cpu_large,
+    'eiger:mc': eiger_cpu_large,
+    'pilatus:mc': pilatus_cpu_large,
+
+}
+
+
+class LAMMPSCheck(LAMMPS):
+    modules = ['LAMMPS']
+    strict_check = False
+    extra_resources = {
+        'switches': {
+            'num_switches': 1
+        }
+    }
+
+    tags = {'scs', 'external-resources'}
+    maintainers = ['TR', 'VH']
+    energy_value = -4.6195
+    energy_tolerance = 6.0E-04
+
+    @run_after('setup')
+    def set_generic_perf_references(self):
+        self.reference.update({'*': {
+            self.benchmark: (0, None, None, 'timesteps/s')
+        }})
+
+    @run_after('setup')
+    def set_perf_patterns(self):
+        self.perf_patterns = {
+            self.benchmark: sn.extractsingle(r'\s+(?P<perf>\S+) timesteps/s',
+                                             self.stdout, 'perf', float)
+        }
+
+    @run_after('init')
+    def source_install(self):
+        # Reset sources dir relative to the SCS apps prefix
+        self.sourcesdir = os.path.join(self.current_system.resourcesdir,
+                                       'LAMMPS')
+
+    @run_after('init')
+    def env_define(self):
         if self.current_system.name in ['eiger', 'pilatus']:
             self.valid_prog_environs = ['cpeGNU']
         else:
             self.valid_prog_environs = ['builtin']
-        self.modules = ['LAMMPS']
 
-        # Reset sources dir relative to the SCS apps prefix
-        self.sourcesdir = os.path.join(self.current_system.resourcesdir,
-                                       'LAMMPS')
-        energy_reference = -4.6195
-        energy = sn.extractsingle(
-            r'\s+500000(\s+\S+){3}\s+(?P<energy>\S+)\s+\S+\s\n',
-            self.stdout, 'energy', float)
-        self.perf_patterns = {
-            'perf': sn.extractsingle(r'\s+(?P<perf>\S+) timesteps/s',
-                                     self.stdout, 'perf', float),
-        }
-        energy_diff = sn.abs(energy-energy_reference)
-        self.sanity_patterns = sn.all([
-            sn.assert_found(r'Total wall time:', self.stdout),
-            sn.assert_lt(energy_diff, 6e-4)
-        ])
-        self.strict_check = False
-        self.extra_resources = {
-            'switches': {
-                'num_switches': 1
-            }
-        }
+    @run_after('init')
+    def set_tags(self):
+        self.tags |= {'maintenance' if self.benchmark == 'maint'
+                      else 'production'}
+
+
+@rfm.simple_test
+class LAMMPSGPUCheck(LAMMPSCheck):
+    scale = parameter(['small', 'large'])
+    benchmark = parameter(['prod', 'maint'])
+    valid_systems = ['daint:gpu']
+    executable = 'lmp_mpi'
+    input_file = 'in.lj.gpu'
```

Comment:
I think that input files should be parameters from the base class in the library.